lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -104,8 +105,9 @@ export class PublicClientApplication {
      * Use when initiating the login process by redirecting the user's browser to the authorization endpoint.
      * @param {@link (AuthenticationParameters:type)}
      */
-    loginRedirect(request: msalAuth.AuthenticationParameters): msalAuth.TokenResponse {
-        throw new Error("Method not implemented.");
+    async loginRedirect(request: AuthenticationParameters): Promise<TokenResponse> {
```

Comment:
I thought we decided `loginRedirect` would not return a promise, as it is misleading?

Reply:
Currently the helpers and other functions have asynchronous steps, such as generating the PKCE codes or waiting for the endpoint discovery to complete. Since we were moving towards async/await I continued to use this pattern, which requires loginRedirect to be async. If we want to change how this works, I am ok with it, but it will mean that we are using the .then/.catch in loginRedirect instead of async/await.

Reply:
Ah, right. How about a private async loginRedirect method, which the public method invokes synchronously?

Reply:
Hm ok, let me see what I can try.